RUFAS/input_manager.py

```diff
@@ -53,120 +76,285 @@ def _load_metadata(self, metadata_path: str = "input/example_metadata.json") ->
+        try:
+            with open(file_path, "r") as csv_file:
+                data_frame = pd.read_csv(csv_file)
+                data_dict = {column: data_frame[column].tolist() for column in data_frame.columns}
+                if not data_frame.empty:
+                    om.add_log("load_data_successful",
+                               f"Successfully loaded data from {file_path}.",
+                               info_map)
+                return data_dict
+        except Exception as e:
+            raise e
 
-    def _validate_data(self, eager_termination: bool = True) -> bool:
+    def _validate_data(self, eager_termination: bool) -> bool:
         """
-        Validates input data and attempts to fix any invalid input data.
+        Validates data from input files and adds data to pool if valid.
 
         Parameters
         ----------
-        eager_termination : bool, default=True
-            If true, the process will be terminated upon finding invalid data.
+        eager_termination : bool
+            If True, the process will be terminated as soon as finding invalid data and failing to fix it.
+            If False, the process will be terminated after going through and validating the entire data,
+            if invalid data is found.
 
         Returns
         -------
         bool
-            True if all data is valid; False otherwise.
+            True if data is valid, otherwise False.
         """
         info_map = {"class": self.__class__.__name__,
                     "function": self._validate_data.__name__,
                     }
-        valid_elements_counter = 0
-        invalid_elements_counter = 0
-        fixed_elements_counter = 0
-        invalid_critical_elements_counter = 0
-        total_elements_checked_counter = 0
-
-        for key in self.__pool.keys():
-            for variable, value in self.__pool[key].items():
-                total_elements_checked_counter += 1
-                if self._validate_element(variable, value):
-                    valid_elements_counter += 1
+        valid_items_counter = 0
+        invalid_critical_items_counter = 0
+        total_items_counter = 0
+
+        for module_key, details in self.__metadata["files"].items():
+            file_path = details["path"]
+            if details["type"] == "json":
+                data = self._load_data_from_json(file_path)
+            elif details["type"] == "csv":
+                data = self._load_data_from_csv(file_path)
+            else:
+                om.add_warning("InputManager load data file is not csv/json",
+                               f"{module_key} data must be available in either csv or json file type.",
+                               info_map)
+                continue
+
+            property_map_key = details["properties"]
+            module_properties = self.__metadata["properties"][property_map_key]
+            for element in module_properties.keys():
```

Comment:
nit. when iterating through an iterable with plural name, individual items in it are singular:
```suggestion
            for property in module_properties.keys():
```

Reply:
Updated.